### PR TITLE
MINOR: Setup linking between PR and Issue in GitHub

### PR DESCRIPTION
**Pull request by amol-**

### CODE REVIEW

**Review by assignUser — APPROVED** [review]
LGTM  :+1: We can add/refine functionality later on if needed.

**Comment by jorisvandenbossche** [comment]
If we update the issue body, can we leave out the comment? Because then that is just duplicated (and in general the bot comments, if they are just informational without requiring action, are a bit notification noise)

File: .github/workflows/dev_pr/link.js [hunk 1]
```diff
@@ -82,11 +82,22 @@ async function commentJIRAURL(github, context, pullRequestNumber, jiraID) {
...
+    const pullInfo = await github.pulls.get({
+        pull_number: pullRequestNumber,
+        owner: context.repo.owner,
+        repo: context.repo.repo,
+    }).data;
+    await github.pulls.update({
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      pull_number: pullRequestNumber,
+      body: pullInfo.body + "\n" + message;
```
Comment (raulcd) [inline .github/workflows/dev_pr/link.js:99]: we can retrieve the body from the context:
```suggestion
      body: context.payload.pull_request.body + "\n" + message
```
and avoid doing the request for the PR. I've tested this on my fork and the only annoying thing is that we might want to deal with the body being null, see here: https://github.com/raulcd/arrow/pull/56#issue-1467688003

**Comment by amol-** [comment]
> If we update the issue body, can we leave out the comment? Because then that is just duplicated (and in general the bot comments, if they are just informational without requiring action, are a bit notification noise)

Agree, made https://github.com/apache/arrow/issues/14790 to track the improvement suggestion.

**Review by raulcd — APPROVED** [review]
+1

**Pull request merged**